src/anyio/_core/_subprocesses.py

```diff
@@ -84,7 +87,7 @@ async def drain_stream(stream: AsyncIterable[bytes], index: int) -> None:
 
     async with await open_process(
         command,
-        stdin=PIPE if input else DEVNULL,
+        stdin=(input and PIPE) or stdin or DEVNULL,
```

Comment:
I think this is tricky to quickly parse correctly, I'd go with a super explicit if-statement above:

```python
if input:
   open_process_stdin = PIPE
elif stdin:
   open_process_stdin = stdin
else:
   open_process_stdin = NEVNULL
```

Reply:
I pushed some changes to this PR. What do you think?

Reply:
even better :+1: